tests/unit/amo/reducers/test_recommendations.js

```diff
@@ -0,0 +1,84 @@
+import reducer, {
+  abortFetchRecommendations,
+  fetchRecommendations,
+  getRecommendationsByGuid,
+  initialState,
+  loadRecommendations,
+} from 'amo/reducers/recommendations';
+import { createInternalAddon } from 'core/reducers/addons';
+import { createStubErrorHandler } from 'tests/unit/helpers';
+import { fakeAddon } from 'tests/unit/amo/helpers';
+
+
+describe(__filename, () => {
+  it('initializes properly', () => {
+    const state = reducer(undefined, {});
+    expect(state).toEqual(initialState);
+  });
+
+  it('ignores unrelated actions', () => {
+    const state = reducer(initialState, { type: 'UNRELATED_ACTION' });
+    expect(state).toEqual(initialState);
+  });
+
+  it('sets the loading flag when fetching recommendations', () => {
+    const guid = 'some-guid';
+    const state = reducer(undefined, fetchRecommendations({
+      errorHandlerId: createStubErrorHandler().id,
+      guid,
+      recommended: true,
+    }));
+
+    expect(state.byGuid[guid].loading).toEqual(true);
+    expect(state.byGuid[guid].addons).toEqual(undefined);
+  });
+
+
+  it('loads recommendations', () => {
+    const addons = [fakeAddon, fakeAddon];
+    const fallbackReason = 'timeout';
+    const guid = 'some-guid';
+    const outcome = 'recommended_fallback';
+    const state = reducer(undefined, loadRecommendations({
+      addons,
+      fallbackReason,
+      guid,
+      outcome,
+    }));
+
+    const expectedAddons = addons.map((addon) => createInternalAddon(addon));
+
+    const loadedRecommendations = getRecommendationsByGuid({ guid, state });
+
+    expect(loadedRecommendations).not.toEqual(null);
```

Comment:
I understand this line but it is (implicitly) covered by the line below no?

Reply:
True. I'll remove it.